src/Symfony/Component/Lock/Factory.php

```diff
@@ -1,54 +0,0 @@
-<?php
-
-/*
- * This file is part of the Symfony package.
- *
- * (c) Fabien Potencier <fabien@symfony.com>
- *
- * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
- */
-
-namespace Symfony\Component\Lock;
-
-use Psr\Log\LoggerAwareInterface;
-use Psr\Log\LoggerAwareTrait;
-use Psr\Log\NullLogger;
-
-/**
- * Factory provides method to create locks.
- *
- * @author Jérémy Derussé <jeremy@derusse.com>
- *
- * @deprecated "Symfony\Component\Lock\Factory" is deprecated since Symfony 4.4 and will be removed in 5.0 use "Symfony\Component\Lock\LockFactory" instead
- */
-class Factory implements LoggerAwareInterface
-{
-    use LoggerAwareTrait;
-
-    private $store;
-
-    public function __construct(StoreInterface $store)
-    {
-        $this->store = $store;
-
-        $this->logger = new NullLogger();
```

Comment:
Do you know what this is meant to achieve? it seems the Lock creates it own NullLogger by default: https://github.com/symfony/symfony/blob/48ab1da0b4562dc458f0f32e0f1d4e0801c7af32/src/Symfony/Component/Lock/Lock.php#L52

Reply:
The factory implements the `LoggerAwareInterface`. So this was probably meant as a fallback to not have to check for `null` when the actual lock is created later.